rfcs/convergence/array-slots.md

```diff
@@ -0,0 +1,165 @@
+RFC: Array slots
+
+---
+
+@smhigley
+
+## Summary
+
+This RFC proposes modifying the Slots API to accept arrays of slot props to support components like Combobox or List, where the most ergonomic authoring experience is passing in an array of data rather than unique JSX children.
+
+## Problem Statement
+
+So far, all slot usage in vNext components assumes the desired authoring experience is to define slots or children one at a time. That makes sense for the components we've created so far, since none of them render a set of elements that are usually generated from an array of data. Even for Menu, which contains a list of MenuItem children, it's not common for authors to want to convert an array of repetitive data into menu items.
+
+In contrast, Combobox and Listbox patterns do commonly take an array of data and use it to render mostly uniform items. The same is likely going to be true of components like List (if we have one) and DetailsList/Grid. For those controls, good authoring ergonomics lean more towards this:
+
+```js
+<Combobox options={['apple', 'banana', 'cucumber', ...etc]} />
+```
+
+Than this:
+
+```js
+<Combobox>
+  <Option>apple</Option>
+  <Option>banana</Option>
+  <Option>cluster fig</Option>
+  [...etc]
+</Combobox>
```

Comment:
This example is not realistic, this comparison will be more accurate for comparison:

```tsx
const options = ["apple", "banana", "cucumber"];

function App() {
  return (
    <>
      <Combobox options={options} />
      {/* VS. */}
      <Combobox>
        {items.map(item => (
          <Option key={item}>{item}</Option>
        ))}
      </Combobox>
    </>
  );
}
```

https://reactjs.org/docs/lists-and-keys.html#basic-list-component